Userland/Utilities/grep.cpp

```diff
@@ -195,25 +197,28 @@ ErrorOr<int> serenity_main(Main::Arguments args)
 
         auto handle_file = [&matches, binary_mode, suppress_errors, count_lines, quiet_mode,
                                user_specified_multiple_files, &matched_line_count, &did_match_something](StringView filename, bool print_filename) -> bool {
-            auto file = Core::File::construct(filename);
-            if (!file->open(Core::OpenMode::ReadOnly)) {
+            auto file_or_error = Core::Stream::File::open(filename, Core::Stream::OpenMode::Read);
+            if (file_or_error.is_error()) {
                 if (!suppress_errors)
-                    warnln("Failed to open {}: {}", filename, file->error_string());
+                    warnln("Failed to open {}: {}", filename, file_or_error.release_error());
                 return false;
             }
 
-            auto file_size_or_error = Core::File::size(filename);
-            if (file_size_or_error.is_error()) {
-                if (!suppress_errors)
-                    warnln("Failed to retrieve size of {}: {}", filename, strerror(file_size_or_error.error().code()));
-                return false;
-            }
+            auto file = file_or_error.release_value();
+            auto buffered_file = Core::Stream::BufferedFile::create(move(file)).release_value_but_fixme_should_propagate_errors();
 
-            auto file_size = file_size_or_error.release_value();
+            for (size_t line_number = 1; buffered_file->can_read_line().release_value_but_fixme_should_propagate_errors(); ++line_number) {
+                auto buffer = FixedArray<u8>::must_create_but_fixme_should_propagate_errors(PAGE_SIZE);
+
+                auto line_or_error = buffered_file->read_line(buffer.span());
+                if (line_or_error.is_error()) {
+                    if (!suppress_errors)
+                        warnln("Failed to read {}: {}", filename, line_or_error.release_error());
+                    return false;
+                }
+                auto line = line_or_error.release_value();
 
-            for (size_t line_number = 1; file->can_read_line(); ++line_number) {
-                auto line = file->read_line(file_size);
-                auto is_binary = memchr(line.characters(), 0, line.length()) != nullptr;
+                auto is_binary = memchr(line.characters_without_null_termination(), 0, line.length()) != nullptr;
```

Comment:
`DeprecatedString` has a `contains` method, and I think it also works with null bytes.

Reply:
It does indeed work with null bytes.